modules/nw-ipfailover-configuration.adoc

```diff
@@ -0,0 +1,180 @@
+          exec:
+            command:
+            - pgrep
+            - keepalived
+      volumes:
+      - name: lib-modules
+        hostPath:
+          path: /lib/modules
+      - name: host-slash
+        hostPath:
+          path: /
+      - name: etc-sysconfig
+        hostPath:
+          path: /etc/sysconfig
+      # config-volume contains the check script
+      # created with `oc create configmap keepalived-checkscript --from-file=mycheckscript.sh`
+      - configMap:
+          defaultMode: 0755
+          name: keepalived-checkscript
+        name: config-volume
+      imagePullSecrets:
+        - name: openshift-pull-secret <13>
+----
+<1> The name of the IP failover deployment.
+<2> The list of IP address ranges to replicate. This must be provided. For example, `1.2.3.4-6,1.2.3.9`.
+<3> The number of groups to create for VRRP. If not set, a group is created for each virtual IP range specified with the `OPENSHIFT_HA_VIP_GROUPS` variable.
+<4> The interface name for IP failover uses to send VRRP traffic. By default, `eth0` is used.
+<5> The IP failover pod tries to open a TCP connection to this port on each VIP. If connection is established, the service is considered to be running. If this port is set to `0`, the test always passes. The default value is `80`.
+<6> The offset value used to set the virtual router IDs. Using different offset values allows multiple IP failover configurations to exist within the same cluster. The default offset is `0`, and the allowed range is `0` through `255`.
+<7> The number of replicas to create. This must match `spec.replicas` value in IP failover deployment configuration. The default value is `2`.
+<8> The name of the `iptables` chain to automatically add an `iptables` rule to allow the VRRP traffic on. If the value is not set, an `iptables` rule is not added. If the chain does not exist, it is not created, and Keepalived operates in unicast mode. The default is `INPUT`.
+<9> The full path name in the pod file system of a script that is run whenever the state changes.
+<10> The full path name in the pod file system of a script that is periodically run to verify the application is operating.
+<11> The strategy for handling a new higher priority host. The default value is `preempt_delay 300`, which causes a Keepalived instance to take over a VIP after 5 minutes if a lower-priority master is holding the VIP.
+<12> The period, in seconds, that the check script is run. The default value is `2`.
+<13> Create the pull secret before creating the deployment, otherwise you will get an error when creating the deployment.
++
+.Example service YAML for IP failover configuration
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: ipfailover-keepalived-service
+spec:
+  ports:
+    - port: 1985
+      targetPort: 1985
+      name: todo
+    - port: 112
+      targetPort: 112
+      name: vrrp
+  selector:
+    ipfailover: hello-openshift
+  externalIPs:
+  - 1.1.1.1
+  - 1.1.1.2
+----
```

Comment:
I did some experimentation with the configuration, and this service for ipfailover doesn't actually do anything, other than maybe reserving the external IPs. It can be removed without any issue.

Reply:
I'm commenting it out just in case it's needed later. Thank you!